tests/schema/tests.py

```diff
@@ -928,6 +928,39 @@ class Meta:
         self.assertEqual(obj.generated, "foo%")
         self.assertIs(obj.contains_foo, True)
 
+    @isolate_apps("schema")
+    @skipUnlessDBFeature("supports_stored_generated_columns")
+    def test_alter_generated_field(self):
+        class GeneratedFieldIndexedModel(Model):
+            text = TextField(default="foo")
```

Comment:
I'd use `IntegerField` to avoid issues with collations, `BLOB`s, not supported indexes on `TextField` etc. 
```suggestion
            text = IntegerField(default=1)
```